api/app/Models/User.php

```diff
@@ -186,6 +186,16 @@ public function roles()
         return new UserRoleRelation((new Role())->newQuery(), $this, 'roles', 'user_id', 'role_key', 'roles');
     }
 
+    public function bookmarkedArticles()
+    {
+        return $this->morphedByMany(Article::class, 'bookmarkable', 'bookmarks');
+    }
+
+    public function ratedArticles()
+    {
+        return $this->morphedByMany(Article::class, 'rateable', 'ratings');
```

Comment:
thought of this, but you defined pivot for other relation (which was not possible), so decoded not to add yet